src/vs/workbench/contrib/terminalContrib/stickyScroll/browser/terminalStickyScrollContribution.ts

```diff
@@ -91,6 +92,15 @@ export class TerminalStickyScrollContribution extends Disposable implements ITer
 				});
 			}
 		}
+
+		const capability = this._ctx.instance.capabilities.get(TerminalCapability.CommandDetection);
+		if (capability && !this._richCommandDetectionListeners.value) {
+			this._richCommandDetectionListeners.value = capability.onSetRichCommandDetection(() => {
+				this._refreshState();
+			});
+		} else if (!capability) {
+			this._richCommandDetectionListeners.clear();
+		}
```

Comment:
This should live inside `_tryEnable` where command detection is definitely available.